src/WebUI/Pods/PodsActionsCreator.ts

```diff
@@ -18,16 +18,29 @@ export class PodsActionsCreator extends ActionCreatorBase {
         this._actions = ActionsHubManager.GetActionsHub<PodsActions>(PodsActions);
     }
 
-    public getPods(kubeService: IKubeService, labelSelector?: string): void {
-        kubeService && kubeService.getPods(labelSelector || undefined).then(podList => {
-            this._extendPodMetadataInList(podList);
-            if (labelSelector) {
-                this._actions.podsFetchedByLabel.invoke({ podsList: podList, labelSelector: labelSelector, isLoading: false });
-            }
-            else {
-                this._actions.podsFetched.invoke({ podsList: podList, isLoading: false });
-            }
-        });
+    public getPods(kubeService: IKubeService, labelSelector?: string, fetchByLabel: boolean = false): void {
+        if (fetchByLabel && !labelSelector) {
+            // For service's associated pods, fetchByLabel is true; In this scenario if no label selector is supplied, it implies zero pods
+            const podList: V1PodList = {
+                apiVersion: "",
+                items: [],
+                kind: "",
+                metadata: {} as V1ListMeta
+            };
+            // Calling action inside timeout to ensure the action listener is initialized in the ServiceDetails
+            setTimeout(() => this._actions.podsFetchedByLabel.invoke({ podsList: podList, labelSelector: "", isLoading: false }));
```

Comment:
ideally this should not the case, in servicedetails, if we have the following code post getstore code, that will take care of this. just a thought

this._servicesStore.addListener(ServicesEvents.ServicePodsFetchedEvent, this._onPodsFetched);

Reply:
I tried adding listener immediately after getStore, but that does not work. So calling action from timeout.